index.js

```diff
@@ -50,27 +50,42 @@
 			function cb(e, r) {
 				err = e;
 				res = r;
-				done = true;		
+				done = true;
 			}
 		}
 	}
-	
+
 	module.exports = deasync;
-	
+	module.exports.default = deasync;
+
 	module.exports.sleep = deasync(function(timeout, done) {
 		setTimeout(done, timeout);
 	});
-	
+
 	module.exports.runLoopOnce = function(){
 		process._tickCallback();
 		binding.run();
 	};
-	
+
 	module.exports.loopWhile = function(pred){
 	  while(pred()){
 		process._tickCallback();
 		if(pred()) binding.run();
 	  }
 	};
 
+	module.exports.await = function(pr) {
+		var done, result;
+
+		done = false;
+		result = undefined;
+
+		pr.then(function(r) {
+			done   = true;
+			return result = r;
+		});
```

Comment:
needs to handle the onrejected part of a promise, or else this will never complete.